src/TestRunner.lua

```diff
@@ -123,40 +116,52 @@ function TestRunner.runPlanNode(session, planNode, lifecycleHooks)
 
 	lifecycleHooks:pushHooksFrom(planNode)
 
-	for _, childPlanNode in ipairs(planNode.children) do
-		session:pushNode(childPlanNode)
+	local halt = false
+	for _, hook in pairs(lifecycleHooks:getBeforeAllHooks()) do
+		local success, errorMessage = runCallback(hook, "beforeAll hook: ")
+		if not success then
+			session:addDummyError("beforeAll", errorMessage)
+			halt = true
+		end
+	end
 
-		if childPlanNode.type == TestEnum.NodeType.It then
-			if session:shouldSkip() then
-				session:setSkipped()
-			else
-				local success, errorMessage = runNode(childPlanNode)
+	if not halt then
+		for _, childPlanNode in ipairs(planNode.children) do
+			session:pushNode(childPlanNode)
 
-				if success then
-					session:setSuccess()
+			if childPlanNode.type == TestEnum.NodeType.It then
+				if session:shouldSkip() then
+					session:setSkipped()
 				else
-					session:setError(errorMessage)
+					local success, errorMessage = runNode(childPlanNode)
+
+					if success then
+						session:setSuccess()
+					else
+						session:setError(errorMessage)
+					end
+				end
+			elseif childPlanNode.type == TestEnum.NodeType.Describe then
+				TestRunner.runPlanNode(session, childPlanNode, lifecycleHooks)
+
+				-- Did we have an error trying build a test plan?
+				if childPlanNode.loadError then
+					local message = "Error during planning: " .. childPlanNode.loadError
+					session:setError(message)
+				else
+					session:setStatusFromChildren()
 				end
 			end
-		elseif childPlanNode.type == TestEnum.NodeType.Describe then
-			TestRunner.runPlanNode(session, childPlanNode, lifecycleHooks)
-
-			-- Did we have an error trying build a test plan?
-			if childPlanNode.loadError then
-				local message = "Error during planning: " .. childPlanNode.loadError
-				session:setError(message)
-			else
-				session:setStatusFromChildren()
-			end
-		end
 
-		session:popNode()
+			session:popNode()
+		end
 	end
 
 	for _, hook in pairs(lifecycleHooks:getAfterAllHooks()) do
-		runCallback(hook, "afterAll hook: ")
-		-- errors in an afterAll hook are currently not caught
-		-- or attributed to a set of child nodes
+		local success, errorMessage = runCallback(hook, "afterAll hook: ")
+		if not success then
+			session:addDummyError("afterAll", errorMessage)
+		end
 	end
```

Comment:
Do we still want to run `afterAll` when there were errors in `beforeAll`? Is this behavior we should pin?

Reply:
I think we should still run it, since its point is to do cleanup, and you often still need that even if `beforeAll` failed. However, it deserves some consideration before committing to that.

Reply:
Yeah, I think we should keep this. I can't find any documentation on how jest handles this specific case, but I can easily imagine when you'd want this behavior (set up 3 things, but the last fails, you still need to clean up the other two), but I can't think of any reason not to have it.

We should also document the behavior that a failing beforeAll will stop any further code in that block except for the `*All` blocks at that specific level.

Reply:
Doesn't look like we have any in-depth documentation to add this to at the moment. I'll file that as a separate issue.